### PR TITLE
Droth 684 speed limit asset to complementary links

### PR DESCRIPTION
**Pull request by joanacoelho**

### CODE REVIEW

File: digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/Digiroad2ApiSpec.scala [hunk 1]
```diff
@@ -72,18 +72,26 @@ class Digiroad2ApiSpec extends AuthenticatedApiSpec with BeforeAndAfter {
...
+    .thenReturn((vvhRoadlinksForBoundingBox.map(toRoadLink), Nil))
   when(mockRoadLinkService.getRoadLinkFromVVH(1611071l))
     .thenReturn(Some(toRoadLink(VVHRoadlink(1611071l, 91,  List(Point(0.0, 0.0), Point(117.318, 0.0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers))))
   when(mockRoadLinkService.getRoadLinkFromVVH(1611353))
     .thenReturn(Some(toRoadLink(VVHRoadlink(1611353, 235, roadLinkGeometry, Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers))))
-  when(mockRoadLinkService.fetchVVHRoadlinks(Set(1611071l, 1611070l, 1611069l)))
-    .thenReturn(Seq(VVHRoadlink(1611071l, 91,  List(Point(0.0, 0.0), Point(117.318, 0.0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers),
-      VVHRoadlink(1611070l, 91,  List(Point(117.318, 0.0), Point(127.239, 0.0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers),
-      VVHRoadlink(1611069l, 91,  List(Point(127.239, 0.0), Point(146.9, 0.0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers)))
+//  when(mockRoadLinkService.fetchVVHRoadlinks(Set(1611071l, 1611070l, 1611069l)))
```
Comment (joanacoelho) [inline digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/Digiroad2ApiSpec.scala]: This comments may need to be removed

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala [hunk 2]
```diff
@@ -913,34 +914,34 @@ class OracleLinearAssetDao(val vvhClient: VVHClient) {
...
   }
 
   /**
     * Updates speed limit value. Used by SpeedLimitService.updateValues, SpeedLimitService.split and SpeedLimitService.separate.
     */
   def updateSpeedLimitValue(id: Long, value: Int, username: String, municipalityValidation: Int => Unit): Option[Long] = {
-    def validateMunicipalities(vvhLinks: Seq[(Long, Double, Seq[Point], Int)]): Unit = {
-      vvhLinks.foreach(vvhLink => municipalityValidation(vvhLink._4))
-    }
+    //def validateMunicipalities(vvhLinks: Seq[(Long, Double, Seq[Point], Int, LinkGeomSource)]): Unit = {
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDao.scala]: This comments may need to be removed

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/LinearAssetServiceSpec.scala [hunk 3]
```diff
@@ -32,7 +32,7 @@ class LinearAssetServiceSpec extends FunSuite with Matchers {
   when(mockVVHClient.fetchByLinkIds(any[Set[Long]])).thenReturn(Seq(VVHRoadlink(388562360l, 235, Seq(Point(0, 0), Point(10, 0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers)))
   when(mockVVHClient.fetchRoadLinkOrComplementaryFromVVH(any[Long])).thenReturn(Some(VVHRoadlink(388562360l, 235, Seq(Point(0, 0), Point(10, 0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers)))
 
-
+  //TODO replace the roadlink for roadLinkWithLinkSource
```
Comment (joanacoelho) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/LinearAssetServiceSpec.scala]: Maybe this TODO needs to be checked

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDaoSpec.scala [hunk 4]
```diff
@@ -14,25 +14,32 @@ import org.scalatest.{FunSuite, Matchers, Tag}
...
   private def daoWithRoadLinks(roadLinks: Seq[VVHRoadlink]): OracleLinearAssetDao = {
     val mockVVHClient = MockitoSugar.mock[VVHClient]
 
     when(mockVVHClient.fetchByLinkIds(roadLinks.map(_.linkId).toSet))
       .thenReturn(roadLinks)
 
+    when(mockRoadLinkService.fetchVVHRoadlinksAndComplementary(roadLinks.map(_.linkId).toSet))
+      .thenReturn(roadLinks)
+
+    //when(mockVVHClient.complementaryData.fetchComplementaryRoadlink(roadLinks.map(_.linkId).head))
```
Comment (joanacoelho) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/linearasset/oracle/OracleLinearAssetDaoSpec.scala]: This comment may need to be removed

**Comment by joanacoelho** [comment]
Otherwise it seems ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/SpeedLimitService.scala [hunk 5]
```diff
@@ -109,7 +109,7 @@ class SpeedLimitService(eventbus: DigiroadEventBus, vvhClient: VVHClient, roadLi
     val persistedSpeedLimits = withDynTransaction {
       dao.getSpeedLimitsChangedSince(sinceDate, untilDate)
     }
-    val roadLinks = roadLinkServiceImplementation.getRoadLinksByLinkIdsFromVVH(persistedSpeedLimits.map(_.linkId).toSet)
+    val roadLinks = roadLinkServiceImplementation.getRoadLinksAndCompelemtnaryByLinkIdsFromVVH(persistedSpeedLimits.map(_.linkId).toSet)
```
Comment (oskarmalmstrom) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/SpeedLimitService.scala]: getRoadLinksAnd**_Compelemtnary_**ByLinkIdsFromVVH typo

**Pull request merged**